ocaml/xapi/xapi_pool_update.ml

```diff
@@ -309,7 +309,13 @@ let guidance_from_string = function
 let parse_update_info xml =
   match xml with
   | Xml.Element ("update", attr, children) ->
-      let key = try List.assoc "key" attr with _ -> "" in
+      let key =
+        List.assoc_opt "key" attr |> Option.value ~default:"" |> function
+        | "" ->
+            ""
```

Comment:
Can we make `key` an option type instead? Then we are forced to handle it properly without relying on the special value `""`. Or would that cause too much churn?

Reply:
It doesn't cause too much churn so I'll change it

Unfortunately we don't get too much benefit because the DB still stores the key as a plain string (and when constructing the yum config we get the key from the DB rather than from values of type `update_info`)